pandas/core/reshape/merge.py

```diff
@@ -488,7 +488,15 @@ def __init__(self, left, right, how='inner', on=None,
         self.right_on = com.maybe_make_list(right_on)
 
         self.copy = copy
+
+        if suffixes == (None, None):
+            raise ValueError('Column names cannot be identical after merging.')
```

Comment:
I think you'd be fine just converting `None` to the empty string; if both are `None` it should get caught further down the line, as this is already the existing behavior:
```python
In [2]: df = pd.DataFrame([1])

In [3]: df.merge(df, left_index=True, right_index=True, suffixes=(None, None))
---------------------------------------------------------------------------
ValueError: columns overlap but no suffix specified: RangeIndex(start=0, stop=1, step=1)

In [4]: df.merge(df, left_index=True, right_index=True, suffixes=('', ''))
---------------------------------------------------------------------------
ValueError: columns overlap but no suffix specified: RangeIndex(start=0, stop=1, step=1)
```

This does seem a little inconsistent though, since specifying duplicated non-blank suffixes does not raise:
```python
In [5]: df.merge(df, left_index=True, right_index=True, suffixes=('foo', 'foo'))
Out[5]:
   0foo  0foo
0     1     1
```
So could maybe instead just check if both are identical after the `None`-->`''` conversion and raise here instead?  Could probably save this for a separate PR if it warrants more discussion though.  Not sure if there's a reason we're not raising in one case versus the other.

Reply:
yeah, i noticed this... but also feel wired in this inconsistent behaviour, I could later make another PR to deal with this suffix warrant issue (duplicated non-blank suffixes).
anyway, the change following this comment is added!